docs/migration/3_0.md

```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
+- `storageSync` configuration for `auth` branch in ngrx store was removed. State of token, userId is now synchronized with `AuthStatePersistenceService`. Override this service if you want to sync more properties to localStorage (eg. `refresh_token`).
+
+### Kyma module
+
+Kyma module were removed with all it's code. We expose the same functionality through configuration of auth.
+
+To fetch OpenId token along with access token in Resource Owner Password Flow you have to use following configuration.
+
+``` ts
+authentication: {
+  client_id: 'client4kyma',
+  client_secret: 'secret',
+  OAuthLibConfig: {
+    responseType: 'id_token',
+    scope: 'openid',
+    customTokenParameters: ['token_type', 'id_token'],
+  },
+},
+```
+
+Then you can access OpenId token with `OAuthLibWrapperService.getIdToken` method. For more options related to OpenId token look into `angular-oauth2-oidc` library documentation.
+
+### ASM module refactor
+
+- `getCustomerSupportAgentTokenState`, `getCustomerSupportAgentToken` and `getCustomerSupportAgentTokenLoading` were removed from `AsmSelectors`. To get token use `AuthStorageService.getToken` and `AsmAuthStorageService.getTokenTarget` to check if it belongs to CS agent.
+- Effects in `AsmModule` now instead of `makeErrorSerializable` use `normalizeHttpError` for error transformation.
+- `storageSync` configuration for `asm` branch in ngrx store was removed. State of ASM UI, tokens is now synchronized with `AsmStatePersistenceService`.
+- `CSAGENT_TOKEN_DATA` variable was removed.
+- `AsmState.csagentToken` was removed. Token is now stored in `AuthStorageService`. Check `AsmAuthStorageService.getTokenTarget` to validate if the token belongs to the CS agent.
+- `AsmActions` no longer contains actions related to customer agent token. Instead interact directly with `CsAgentAuthService`.
+- `CustomerSupportAgentTokenInterceptor` interceptor was removed. Token and error handling for CS agent requests are now handled by `AuthInterceptor` and `AsmAuthHttpHeaderService`.
+
+#### AsmAuthService
+
+Service was renamed to `CsAgentAuthService`. `AsmAuthService` is now responsible for making `AuthService` aware of ASM and adjusts it for CS agent support.
+
+- `AsmAuthService.authorizeCustomerSupportAgent` was moved to `CsAgentAuthService`. It now performs full login flow for CS agent and resolves when it completes.
+- `AsmAuthService.startCustomerEmulationSession` was moved to `CsAgentAuthService`. Behavior haven't changed.
+- `AsmAuthService.isCustomerEmulationToken` was removed. To check token use `AuthStorageService.getToken` and to check if it belongs to CS agent use `AsmAuthStorageService.getTokenTarget`.
+- `AsmAuthService.getCustomerSupportAgentToken` was removed. To check token use `AuthStorageService.getToken` and to check if it belongs to CS agent use `AsmAuthStorageService.getTokenTarget`.
+- `AsmAuthService.getCustomerSupportAgentTokenLoading` was moved to `CsAgentAuthService`. Warning! It is not implemented there yet.
```

Comment:
👀

Reply:
There is a ticket for that and we might able to complete that before stable release ;)